### DISCUSSION TITLE
Add when dev has =>5 tickets unique reports, new achievements cannot be added to >= unofficial

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAWeb#702] [Ideas]
The idea here (as suggested in devs) is to get devs to have their ticket count to 0 before they can add new achievements. (if it's found devs are closing tickets without fixing the issue then it will result in discipline.)

This system would need to be automatic.

**Comment by grenade44** [comment]
This is a great idea😀👍

**Comment by meleu** [comment]
I fully agree that devs need to fix their tickets, but I don't agree with the idea proposed here. Maybe I would if the count was something around 5.

**Talking as a dev:**

I have 3 open tickets, all for the same cheevo: http://retroachievements.org/ticketmanager.php?g=574&ampt=1

The cheevo is for not using grenades. One report is that the cheevo popped even with the player using grenades. The other two reports are "doesn't trigger" ones.

I tested it and it works as expected. I also PMed every person listed as having this cheevo and they say it's working fine (alright, this isn't a very precise criterion cause some users love having cheevos popping at a wrong time).

I'm still investigating, but it depends on the help and freetime of the players that had problems with this cheevo.

If I would restricted from creating new cheevos because of my tickets above, I would (1) give up to create new cheevos; or (2) demote that cheevo and close the ticket as a really buggy one.

**Talking as an admin:**

> if it's found devs are closing tickets without fixing the issue then it will result in discipline

It means more task for admins. And judging a dev and maybe removing his dev status... I wouldn't like  to deal with this situation.

**Comment by ScottFromDerby** [comment]
It's an interesting proposal. I can fully understand the thinking behind it. My biggest concern is the part regarding "discipline". Who authorises this discipline? Are you going to kick me off the site because I have open tickets?

**Comment by SyrianBallaS** [comment]
Note: I am not planning on merging this with the master branch.

A lot of this stuff can be done easily in ASP.NET Core using Identity Framework (role and claim-based authorization) where claims are done by policy. Plus the Lumen seems nearly identical to how .NET Core is set up. Mainly just as a POC, until the seeding happens. Plus it allows MVC and WebAPI to work in tandem.

I'm probably gonna make netcoreapp version of this (works with mysql) it uses Kestrel as a Web Server which is based off of libuv from Linux. Or as a reverse proxy with IIS/Apache/etc. 

It uses sqllite as the db by default but mysql should be fine.

**Comment by kdecks** [comment]
@ScottFromDerby  @meleu  discipline would only be for those who are found to close tickets without actually resolving the issue, specifically if they are found to be being careless. Consequences would be a warning, and then on another occurrence loss of developer status. 

This discipline would be an administrative action, and would be expected to be a rare occurrence.
 
 Besides discipline, how else could we prevent developers from closing tickets without resolving issues? Preferably automatically.

 I have a few Open tickets as well, but requiring tickets to be closed would encourage discipline and responsibilities without anyone breathing down their backs.

**Comment by kdecks** [comment]
I think a minimum allowed count of 5 is a reasonable, balanced approach, that way someone would not be locked out mid work in most situations.

**Comment by meleu** [comment]
@kdecks I think you pinged another Scott :lel:
"Our" Scott is @ScottFromDerby :wink: 

Regarding numbers, I agree that 5 is reasonable. But we need to discuss about duplicated reports. IIRC last time we both talked about it we have not reached a consensus (IMO opinion there's no need to keep opened several tickets about the same issue).

**Comment by kdecks** [comment]
Fixed.

My concern about closing duplicates would be resolved if there was a **duplicate (closed)** ticket status.

At present you'll see **resolved** which implies that it's ... been resolved. (Or **closed**, which is only happens when you demote.) Seeing many **resolved** tickets on an achievement implies that it's been fixed, has been worked on, or that there is a common user mistake.

**Comment by kdecks** [comment]
The limit needs to considered 5 unique reporters (not 5 tickets), as often reporters will dump a big batch of tickets at once.

**Comment by Keltron3030** [comment]
My proposal to approach this problem would be an open ticket % column in the developer stats feature.  If you're over 5%(or other agreed upon rate) you lose dev status until you bring the ratio down. 


Regarding duplicate tickets; for example if you have 10 tickets for one achievement, this just emphasizes that you've created for a popular set and you need to get it fixed before you turn off many users.

The top of the current list may be more telling if we had this % column.  There could be many active devs with a relatively high # of tickets for a low number of achievements created. 

A % would be good to keep an eye on the newer developers to correct bad practices before they become pervasive.

I think this approach is preferred vs. a flat number of tickets since it is dynamic and will not penalize those developers with a large number of achievements which take more effort to maintain.

The site tends to reward the most prolific devs with quantity stats, but there's no quality stat.  This feature would give the added benefit of seeing this stat on the user page as an indicator of a developer who also holds quality of work in high esteem.

**Comment by kdecks** [comment]
I've given it thought.
I second that the ticket percent stat would be a useful analytic for the Dev stats page and the user page.

*This is different than previously proposed:* Here's what I believe would be the simplest approach to this problem and would best create what we want:

> If a developer had 5 or more tickets, they would not be able to publish new achievements to core from unofficial.
> 
> The would still be able to publish to unofficial, and to make changes to achievement in core, as that is needed to fix tickets. Their developer role title would be unaffected.

**Comment by kdecks** [comment]
The difference from previously suggested it to still restrict devs, if they have multiple tickets from the same player. (For simplicity's sake)

And previously it was suggested to not allow publishing into unofficial, where now I'm suggesting that would still be allowed. This seems to be a good motivator for JrDevs to get their work "up to code," and should probably work fine for motivated developers.

Also, I'm against giving a much bigger allowance for devs that have published many more achievements. I don't see the benefit.

**Comment by Jamiras** [comment]
While not strictly enforced, #1027 warns the user they have tickets when they try to make a claim.
> If the user [...] has open tickets the following lines will be displayed on the confirmation popup as well.
> `Please ensure any open tickets have been addressed before making this claim.`
